homeassistant/components/air_pollutants/nilu.py

```diff
@@ -0,0 +1,265 @@
+
+ATTR_AREA = 'area'
+ATTR_POLLUTION_INDEX = 'nilu_pollution_index'
+
+CONF_ALLOWED_AREAS = [
+    'Ålesund',
+    'Zeppelinfjellet',
+    'Tustervatn',
+    'Trondheim',
+    'Tromsø',
+    'Sør-Varanger',
+    'Stavanger',
+    'Sarpsborg',
+    'Sandve',
+    'Prestebakke',
+    'Oslo',
+    'Narvik',
+    'Moss',
+    'Mo i Rana',
+    'Lørenskog',
+    'Lillestrøm',
+    'Lillesand',
+    'Lillehammer',
+    'Kårvatn',
+    'Kristiansand',
+    'Karasjok',
+    'Hurdal',
+    'Harstad',
+    'Hamar',
+    'Halden',
+    'Grenland',
+    'Gjøvik',
+    'Fredrikstad',
+    'Elverum',
+    'Drammen',
+    'Bærum',
+    'Brumunddal',
+    'Bodø',
+    'Birkenes',
+    'Bergen'
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Inclusive(CONF_LATITUDE, 'coordinates',
+                  'Latitude and longitude must exist together'): cv.latitude,
+    vol.Inclusive(CONF_LONGITUDE, 'coordinates',
+                  'Latitude and longitude must exist together'): cv.longitude,
+    vol.Exclusive(CONF_AREA, 'station_collection',
+                  'Can only configure one specific station or '
+                  'stations in a specific area pr sensor. '
+                  'Please only configure station or area.'
+                  ): vol.All(cv.string, vol.In(CONF_ALLOWED_AREAS)),
+    vol.Exclusive(CONF_STATION, 'station_collection',
+                  'Can only configure one specific station or '
+                  'stations in a specific area pr sensor. '
+                  'Please only configure station or area.'
+                  ): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_SHOW_ON_MAP, default=False): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the NILU air pollutants sensor."""
+    import niluclient as nilu
+    name = config.get(CONF_NAME)
+    area = config.get(CONF_AREA)
+    stations = config.get(CONF_STATION)
+    show_on_map = config.get(CONF_SHOW_ON_MAP)
+
+    sensors = []
+
+    if area:
+        stations = nilu.lookup_stations_in_area(area)
+    elif not area and not stations:
+        latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+        longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+        location_client = nilu.create_location_client(latitude, longitude)
+        stations = location_client.station_names
+
+    for station in stations:
+        client = NiluData(nilu.create_station_client(station))
+        client.update()
+        sensors.append(NiluSensor(client, name, show_on_map))
+
+    add_entities(sensors, True)
+
+
+class NiluData:
+    """Class for handling the data retrieval."""
+
+    def __init__(self, api):
+        """Initialize the data object."""
+        self.api = api
+
+    @property
+    def data(self):
+        """Get data cached in client."""
+        return self.api.data
+
+    @Throttle(SCAN_INTERVAL)
```

Comment:
Sort of, yes. The source only aggregates and publishes new data once an hour, so no need to request the API too often unnecessary. There is no rate limit, but no need to overload them 🙂 A couple of times an hour I believe would be enough to get good refresh time and at the same time protect the source a little bit.

Reply:
btw, is it possible to tell the update to work with an cron instead of an time interval? Then a cron with `5 */1 * * *` would be much better to achieve this.

Reply:
If we want a different interval than a set time interval, we can take control and schedule the interval in the platform.

I don't know by heart what that cron interval means.

Reply:
I'm still a bit green on all the inner workings of HA, so don't know how to do the manual scheduling during setup, but the idea was to do the update with "minute 5 past every hour" as most of the stations seams to publish around the hour, or +/- 5 min.  
But every 30 or 15 min would not mean too much extra requests, and considering extra complexity to scheduling, it might not be worth it?

Reply:
Yeah, I suggest setting it to 1 hour interval.